src/scores/processing.py

```diff
@@ -1,8 +1,188 @@
 """Tools for processing data for verification"""
+import operator
 from typing import Union
 
+import numpy as np
+import pandas as pd
 import xarray as xr
 
+INEQUALITY_MODES = {
+    ">=": (operator.ge, -1),
+    ">": (operator.gt, 1),
+    "<=": (operator.le, 1),
+    "<": (operator.lt, -1),
+}
+# '==' does not map to `operater.eq` and '!=' does not map to operator.ne on purpose.
+# This is because we wish to test within a tolerance.
+EQUALITY_MODES = {"==": (operator.le), "!=": (operator.gt)}
+
+
+def check_binary(data: Union[xr.DataArray, xr.Dataset], name: str):
```

Comment:
Think the type for data should be XarrayLike from scores.typing

Reply:
Updated the function to work with Datasets and added tests, since there was a bug